cmd/influxd/upgrade/upgrade.go

```diff
@@ -219,3 +666,72 @@ func influxDirV1() (string, error) {
 
 	return dir, nil
 }
+
+// homeOrAnyDir retrieves user's home directory, current working one or just none.
+func homeOrAnyDir() string {
+	var dir string
+	u, err := user.Current()
+	if err == nil {
+		dir = u.HomeDir
+	} else if home := os.Getenv("HOME"); home != "" {
+		dir = home
+	} else if home := os.Getenv("USERPROFILE"); home != "" {
+		dir = home
+	} else {
+		wd, err := os.Getwd()
+		if err != nil {
+			dir = ""
+		} else {
+			dir = wd
+		}
+	}
+
+	return dir
+}
+
+func rawDurationToTimeDuration(raw string) (time.Duration, error) {
```

Comment:
I don't see why this is needed when we already have `time.ParseDuration` and the flag package already directly supports `time.Duration`-typed flags.

Reply:
Same as above. Actually, this is copied from the setup command, but it would be best to reuse this. Could you, please, suggest a good place to put it?